src/Plugins/SimplnxCore/src/SimplnxCore/Filters/Algorithms/CropEdgeGeometry.cpp

```diff
@@ -0,0 +1,410 @@
+#include "CropEdgeGeometry.hpp"
+
+#include "simplnx/DataStructure/DataArray.hpp"
+#include "simplnx/DataStructure/DataGroup.hpp"
+#include "simplnx/DataStructure/Geometry/EdgeGeom.hpp"
+#include "simplnx/Utilities/DataArrayUtilities.hpp"
+#include "simplnx/Utilities/ParallelDataAlgorithm.hpp"
+
+using namespace nx::core;
+
+namespace
+{
+/**
+ * @brief
+ * @tparam T
+ */
+template <typename T>
+class CropEdgeGeomArray
+{
+public:
+  CropEdgeGeomArray(const IDataArray& oldCellArray, IDataArray& newCellArray, const AttributeMatrix& srcAttrMatrix, const std::vector<bool>& tupleMask, const std::atomic_bool& shouldCancel)
```

Comment:
I would either remove the shouldCancel arg/member variable or utilize it in the function.